### ISSUE TITLE
cal_return_slist not robust

### ISSUE THREAD

**Issue by mhardcastle** [tammojan/facet-calibration#3]
The current code for cal_return_slist, which returns sources in a square region on the sky, is not robust to special cases in the co-ordinate system (e.g. spanning RA=0h). It's fine for most fields but should be fixed at some point.

**Comment by twshimwell** [comment]
Hi all,

I noticed that if you set atrous_do to False in your peelsourceinfo file (i.e. the entry after the RA and DEC) then it was being set to True in the code because it was not being read as a Boolean. This caused me issues when masking the self calibration images and as a consequence the selfcal was making the data worse rather than better.  I have updated the doDDE_v21_a2256.py to now read this parameter as a Boolean.

Thanks,

Tim

**Comment by mhardcastle** [comment]
Good catch Tim. I had just got used to the fact that it was defaulting to True without worrying too much about whether it should be or not. (In fact I am not convinced that the default shouldn't be True, switching to False if you have problems with artefacts.)

I was about to go in and force it to be True for the full facet clean, where it clearly needs to be true for proper subtraction, but I see Reinout has beaten me to it!

**Comment by rvweeren** [comment]
For the facet imaging atrous_do should be forced to True. That's quite important to pick up extended emission. For the selfcal the default should be False, there was indeed a bug there which resulted in multiscale being always used in the selfcal. Tim's fixed that part that but that then resulted in problems with the facet source detection. Now everything should be working again.

**Comment by rvweeren** [comment]
> > The current code for cal_return_slist, which returns sources in a square region on the sky, is not robust to special cases in the co-ordinate system (e.g. spanning RA=0h). It's fine for most fields but should be fixed at some point.

Martin: Is the code save for high-Dec fields? (I'm working at Dec=80 degr)

**Comment by mhardcastle** [comment]
Safe, probably. cos(dec) is calculated at the reference position, so you will get a non-trivial distortion of the square on the sky for the largest possible fields. but not I think so much as to go over the border and restore sources that then won't be imaged. It's starting to get non-ideal at those high decs though. Implementation of my mask idea would be preferable: happy to try to take a look tomorrow.

**Comment by mhardcastle** [comment]
Ok, an implementation of the mask idea is now in the tree. New code slists_v2.py implements this (actually turns out to be very easy once you figure out the pyrap images incantations needed!) and also combines cal_return_slist and return_slist so that one module can be imported for both and we get rid of some duplicate code. The old code is still there if we need to back off. I get identical results with this on some spot checks to the old code but it would be very interesting if you can try it on your high-dec field, Reinout.

**Comment by rvweeren** [comment]
I'll give it a try next week.
